### PR TITLE
#3049 Avoid dumping 0 instead of undefined values for HVDC power values when applicable

### PR DESCRIPTION
**Pull request by barondim**
…Component()

**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [x] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [x] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp [hunk 1]
```diff
@@ -394,7 +395,41 @@ createBusBreakerNetwork(const BusBreakerNetworkProperty& properties) {
...
+  if (properties.instantiateVscConverterWithDisconnectedHvdc) {
+    vl1.newVscConverterStation()
+        .setId("MyVscConverter")
+        .setName("MyVscConverter_NAME")
+        .setConnectableBus("MyBus")
+        .setLossFactor(3.0)
+        .setVoltageRegulatorOn(true)
+        .setVoltageSetpoint(1.2)
+        .setReactivePowerSetpoint(-1.5)
+        .add();
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp:407]: Why duplicating the code and not just adding the bus depending on the property?

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp [hunk 2]
```diff
@@ -1054,15 +1129,19 @@ TEST(DataInterfaceIIDMTest, testHvdcLineVscConvertersIIDM) {
   hvdc->setValue(HvdcLineInterfaceIIDM::VAR_STATE1, OPEN);
   hvdc->setValue(HvdcLineInterfaceIIDM::VAR_STATE2, OPEN);
   data->exportStateVariablesNoReadFromModel();
-  ASSERT_DOUBLE_EQUALS_DYNAWO(vsc1->getVscIIDM().getTerminal().getP(), -200.);
-  ASSERT_DOUBLE_EQUALS_DYNAWO(vsc1->getVscIIDM().getTerminal().getQ(), -400.);
-  ASSERT_DOUBLE_EQUALS_DYNAWO(vsc2->getVscIIDM().getTerminal().getP(), -600.);
-  ASSERT_DOUBLE_EQUALS_DYNAWO(vsc2->getVscIIDM().getTerminal().getQ(), -800.);
+  ASSERT_DOUBLE_EQUALS_DYNAWO(vsc1->getVscIIDM().getTerminal().getP(), 150.);
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp:1132]: Why this change?

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp [hunk 3]
```diff
@@ -1074,9 +1153,62 @@ TEST(DataInterfaceIIDMTest, testHvdcLineVscConvertersIIDM) {
...
+      false /*instantiateThreeWindingTransformer*/,
+      false /*instantiateBattery*/
+  };
+  shared_ptr<DataInterfaceIIDM> data = createDataItfFromNetwork(createBusBreakerNetwork(properties));
+  exportStateVariables(data);
+  boost::shared_ptr<HvdcLineInterfaceIIDM> hvdc = boost::dynamic_pointer_cast<HvdcLineInterfaceIIDM>(data->findComponent("MyHvdcLine"));
+  boost::shared_ptr<VscConverterInterfaceIIDM> vsc1 = boost::dynamic_pointer_cast<VscConverterInterfaceIIDM>(hvdc->getConverter1());
+  boost::shared_ptr<VscConverterInterfaceIIDM> vsc2 = boost::dynamic_pointer_cast<VscConverterInterfaceIIDM>(hvdc->getConverter2());
+
+  ASSERT_TRUE(std::isnan(vsc1->getVscIIDM().getTerminal().getP()));
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp:1183]: Would it be possible to add a test in which the NaN stays a NaN?

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp (hunk 1 above)
Comment (barondim) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp:407]: In TestDataInterface.cpp, each unit test uses a different network by switching one single attribute in the BusBreakerNetworkProperty struct. I thought creating one new attribute for each new unit test I wanted to add would be good because it keeps the logic of the previous code.

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp (hunk 2 above)
Comment (barondim) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp:1132]: Just before, vsc1 P attribute is set to 150. Since the hvdc isn't connected, the P attribute remains at 150.

Before this task, disconnected HVDC used to have their attributes changed but now it's not the case anymore.

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp (hunk 1 above)
Comment (rosiereflo) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp:407]: yes this is ok, but the code to create the hvdc is mostly copy paste, so it would be better to factorize

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp (hunk 3 above)
Comment (barondim) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp]: Yes it is. I've just added this in 2 new commits.

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp (hunk 1 above)
Comment (barondim) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp]: Ok, I defined 3 functions to factorize the code.

**Review by rosiereflo — APPROVED** [review]
